cdhweb/pages/management/commands/exodus.py

```diff
@@ -274,99 +253,46 @@ def migrate_pages(self, page, parent):
+        imgfile_path = '%s/**/*.*' % settings.MEDIA_ROOT
+        # get filenames for all uploaded files
+        filenames = glob.glob(imgfile_path, recursive=True)
+        # aggregate files by basename to identify files with the same
+        # name in different locations
+        unique_filenames = defaultdict(list)
+        for path in filenames:
+            unique_filenames[os.path.basename(path)].append(path)
 
-    # cached collections used for migrated media
-    collections = {
-        # get root collection so we can add children to it
-        'root': Collection.objects.get(pk=get_root_collection_id())
-    }
+        # check files with the same name in multiple locations
+        for key, val in unique_filenames.items():
+            if len(val) > 1:
+                samefile = filecmp.cmp(val[0], val[1], shallow=False)
+                # if the files are the same
+                if samefile:
+                    # keep the first one and remove the others from the
+                    # list of files to be migrated
+                    extra_copies = val[1:]
 
-    def profile_pages(self):
-        """Exodize all existing Profiles to new Profile model."""
-        # all the fields on Profile that moved to Person will have been handled
-        # by django migrations; we just need to create new Page models
-        people_landing = PeopleLandingPage.objects.first()
-        if not people_landing:
-            return
-        # only create profiles if there's actually text in the old one
-        for profile in OldProfile.objects.exclude(bio=""):
-            person = Person.objects.get(user=profile.user)
-            profile_page = Profile(
-                person=person,
-                title=profile.title,
-                slug=self.convert_slug(profile.slug),
-                image=self.get_wagtail_image(profile.image) if profile.image else self.get_wagtail_image(
-                    profile.thumb) if profile.thumb else None,
-                education=profile.education,
-                bio=json.dumps([
-                    {"type": "migrated", "value": profile.bio},
-                ])
-            )
-            # added as child of people landing page so slugs are correct
-            people_landing.add_child(instance=profile_page)
-            people_landing.save()
-            # if the old profile wasn't published, unpublish the new one
-            if profile.status != CONTENT_STATUS_PUBLISHED:
-                profile_page.unpublish()
+                # if not all the same, identify the largest
+                # (all are variant/cropped versions of the same image)
+                else:
+                    largest_file = None
+                    largest_size = 0
+                    for filepath in val:
+                        size = os.stat(filepath).st_size
+                        if size > largest_size:
+                            largest_size = size
+                            largest_file = filepath
```

Comment:
smart

Reply:
isn't this your code??

Reply:
:😆 is it?! I guess I forgot, it looked new because you moved it around !

Reply:
(not disagreeing though!)